test/e2e/reconciler/cache/replication_test.go

```diff
@@ -251,6 +253,101 @@ func replicateAPIExportNegativeScenario(ctx context.Context, t *testing.T, serve
 	scenario.VerifyReplication(ctx, t, cluster)
 }
 
+// replicateClusterWorkspaceShardScenario tests if a ClusterWorkspaceShard is propagated to the cache server.
+// The test exercises creation, modification and removal of the ClusterWorkspaceShard object.
+func replicateClusterWorkspaceShardScenario(ctx context.Context, t *testing.T, server framework.RunningServer, kcpShardClusterClient clientset.ClusterInterface, cacheKcpClusterClient clientset.ClusterInterface) {
+	// The ClusterWorkspaceShard API is per default only available in the root workspace
+	cluster := tenancyv1alpha1.RootCluster
+	resourceName := "sheriffs.wild.wild.west"
+	scenario := &replicateResourceScenario{resourceName: resourceName, resourceKind: "ClusterWorkspaceShard", server: server, kcpShardClusterClient: kcpShardClusterClient, cacheKcpClusterClient: cacheKcpClusterClient}
+
+	t.Logf("Create source ClusterWorkspaceShard %s/%s on the root shard for replication", cluster, resourceName)
+	scenario.CreateSourceResource(t, func() error {
+		cws := &tenancyv1alpha1.ClusterWorkspaceShard{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: resourceName,
+			},
+			Spec: tenancyv1alpha1.ClusterWorkspaceShardSpec{
+				BaseURL: "https://base.kcp.test.dev",
```

Comment:
I guess it is okay for now, it might break when this test is run on a kcp instance that is shared with other tests.

Reply:
This is something worth avoiding up front. What about using GenerateName instead of Name?

Reply:
I think we need a real address or a way to disable this shard from scheduling. We might try to put something on that shard in a multi shard env.

Maybe for this particular test scenario we should actually use `framework.PrivateKcpServer` ?

Reply:
My gut feeling is that we should build the tests and the business logic in a way that is not sensible to that. If a node is not ready in Kubernetes no new pod gets scheduled onto it. I am wondering whether we would need a similar mechanism in kcp for the shards at some point.
Nevertheless we should forge the multi-shard tests so that the placement is constrained to a shard, which is know to be functional (or not for negative tests). It is good to have as few explicit or implicit dependencies between the test cases.
The alternative is to start an additional kcp server to have a "real address", which may make the tests slower. I would favor the other approach for this reason. What do you think?

Reply:
NACK to private KCP in every case. Why don't we have shard scheduling control?

Reply:
Would it be okay to add an annotation to the shard created by this test that would tell the scheduler to not consider the shard?

I think we would have to update the `isValidShard`(https://github.com/kcp-dev/kcp/blob/main/pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_reconcile_scheduling.go#L229) function to look up for the special annotation.

Reply:
That might be a good approach in the near-term, but do we have any idea if we're going to add some sort of scheduling attributes to the shards for the long-term?

Reply:
I think that by default we will schedule randomly and we will allow for specifying a selector for those that have opinions